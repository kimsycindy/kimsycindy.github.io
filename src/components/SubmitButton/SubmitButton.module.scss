@import '../../styles/variables.scss';

.SubmitButton:hover {
  background-color: lighten($accent, 5%);
}

.SubmitButton:active {
  background-color: lighten($accent, 10%);
}
