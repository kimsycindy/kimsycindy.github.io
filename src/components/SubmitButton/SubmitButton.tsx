import React from 'react'
import cx from 'classnames'

import styles from './SubmitButton.module.scss'
import Button from '../Button'
import { BUTTON_TYPES } from '../Button/constants'

interface Props {
  submitting: boolean
  submittingText: string
  status: {
    ok: boolean
    msg: string
  }
  className?: string
}

const SubmitButton: React.FC<Props> = ({
  submitting,
  submittingText,
  status: { ok, msg },
  className,
  children,
}) => (
  <Button
    className={cx(styles.SubmitButton, className)}
    type={BUTTON_TYPES.SUBMIT}
    disabled={submitting || ok}
  >
    {submitting && submittingText}
    {ok && msg}
    {!submitting && !ok && children}
  </Button>
)

export default SubmitButton
