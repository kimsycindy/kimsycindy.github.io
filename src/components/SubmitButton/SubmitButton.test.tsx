import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'

import SubmitButton from './SubmitButton'

describe('SubmitButton', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <SubmitButton
          submitting={false}
          submittingText="Submitting..."
          status={{ ok: false, msg: 'message' }}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows submittingText on click', () => {
    const submittingText = 'Submitting...'

    const { getByText } = render(
      <SubmitButton
        submitting
        submittingText={submittingText}
        status={{ ok: false, msg: 'message' }}
      />
    )

    expect(getByText(submittingText)).toBeInTheDocument()
  })
})
