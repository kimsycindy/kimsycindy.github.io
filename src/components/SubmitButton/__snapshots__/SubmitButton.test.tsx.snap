// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders correctly 1`] = `
<button
  className="Button SubmitButton"
  disabled={false}
  type="submit"
/>
`;
